### PR TITLE
[TACHYON-394] Modularize playbooks in deploy/vagrant

### PR DESCRIPTION
**Pull request by uronce-cc**
https://tachyon.atlassian.net/browse/TACHYON-394

Modularize playbooks so that it's easier to add new components. see README.md below for the design. 

Tests Run:

<table class="table">
<tr>
    <th>Provider</th><th>Ufs</th><th>Tachyon</th><th>Spark</th>
</tr>
<tr>
    <td>vb</td><td>hadoop1</td><td>release</td><td>release</td>
</tr>
<tr>
    <td>vb</td><td>hadoop1</td><td>local</td><td>none</td>
</tr>
<tr>
    <td>vb</td><td>glusterfs</td><td>github</td><td>none</td>
</tr>
<tr>
    <td>aws</td><td>hadoop2</td><td>github</td><td>github</td>
</tr>
<tr>
    <td>aws</td><td>glusterfs</td><td>github</td><td>none</td>
</tr>
<tr>
    <td>aws</td><td>hadoop1</td><td>local</td><td>none</td>
</tr>
</table>

### CODE REVIEW

File: deploy/vagrant/provision/roles/ufs_hadoop2/files/mkdir.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+if [ ! -d /hadoop ]; then
+	sudo mkdir /hadoop
```
Comment (haoyuan) [inline deploy/vagrant/provision/roles/ufs_hadoop2/files/mkdir.sh]: Please use spaces to replace tab.

**Comment by calvinjia** [comment]
@uronce-cc Thanks for the patch! Its a big improvement over the previous deploy module. What distinction do you use when decided to define shell commands with ansible's `shell` and having a separate script? Is it just if the code is reused?

**Comment by uronce-cc** [comment]
@calvinjia 
Two situations:
1. it's a one-line bash
2. code reuse

File: deploy/vagrant/run_vb.sh [hunk 2]
```diff
@@ -5,11 +5,11 @@ vagrant up --provider=virtualbox --no-provision
...
-    no_color='\033[0m'
-    echo -e ">>> ${purple}visit $master:19999 for Tachyon Web Console${no_color} <<<"
-    echo -e ">>> ${purple}visit $master:8080  for Spark   Web Console${no_color} <<<"
+ HERE=`dirname $0`
+ pushd $HERE > /dev/null
+ master=`tail -n 1 $HERE/files/hosts | cut -d' ' -f1`
+ purple='\033[1;35m'
+ no_color='\033[0m'
+ echo -e ">>> ${purple}visit $master:19999 for Tachyon Web Console${no_color} <<<"
+ echo -e ">>> ${purple}visit $master:8080  for Spark   Web Console${no_color} <<<"
```
Comment (calvinjia) [inline deploy/vagrant/run_vb.sh]: Its kind of weird to have this line if they don't deploy spark, but its not really related to this PR.

**Comment by calvinjia** [comment]
+1 LGTM

I would like to see the bash code in the playbooks all moved to separate files, but thats mostly a style issue that does not need to be addressed in this PR.

**Pull request merged**